coreMacros/src/main/scala/chisel3/internal/sourceinfo/NamingAnnotations.scala

```diff
@@ -118,12 +118,14 @@ class NamingTransforms(val c: Context) {
   def transformModuleBody(stats: List[c.Tree]): Tree = {
     val contextVar = TermName(c.freshName("namingContext"))
     val transformedBody = (new ModuleTransformer(contextVar)).transformTrees(stats)
-
+    // Note: passing "this" to popReturnContext is mandatory for propagation through non-module classes
     q"""
     val $contextVar = $globalNamingStack.pushContext()
     ..$transformedBody
-    $contextVar.namePrefix("")
-    $globalNamingStack.popReturnContext((), $contextVar)
+    if($globalNamingStack.length == 1){
+      $contextVar.namePrefix("")
```

Comment:
This made sense when Module was top-level, since in all cases the first naming prefix you get would correlate directly with something in the Module. In this case, you could have a non-Module class as the top-level, and the first naming prefix would not necessarily correlate with anything in the Module. Is this behavior we want? Or do we want a check somewhere that the naming top-level is the Module?

Reply:
As far as I understand the macro, when top level is not a module what does the transform is harmless and does not reflect in anyway to the emitted names so it is not an issue.